Infrastructure/ContainerBuilder.cs

```diff
@@ -0,0 +1,81 @@
+﻿using BasePlugin.Interfaces;
+using Dal;
+using Dals;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using List;
+using CountDown;
+using DiceRoller;
+using DB.Data;
+using System.Diagnostics;
+using BasePlugin.Services;
+
+namespace Infrastructure
+{
+    public static class Container
+    {
+        public delegate IService ImplementationFactory(RunnerType runner);
```

Comment:
Please change to a meaningful name, something as 'ServiceFactory'
Also, if possible, check the use of Action<RunnerType > instead of delegate

Reply:
I guess there is no alternative to delegate here